code/game/objects/items/holosign_creator.dm

```diff
@@ -90,6 +90,51 @@
 	creation_time = 0
 	max_signs = 3
 
+/obj/item/holosign_creator/powered_shielding
+	name = "Power shield projector"
+	desc = "A holographic projector that creates a Power shield that stops the spread of heat and atmosphere changes"
+	icon_state = "signmaker_powersh"
+	holosign_type = /obj/machinery/holosign/barrier/power_shield/wall
+	creation_time = 0
+	///25 so that you can make a 5x5 room with walls and floors
+	max_signs = 25
+
+/obj/item/holosign_creator/powered_shielding/attack_self(mob/user)
+	toggle_item(user)
+
+/obj/item/holosign_creator/powered_shielding/proc/toggle_item(mob/user)
+	if(holosign_type == /obj/machinery/holosign/barrier/power_shield/wall)
+		holosign_type = /obj/machinery/holosign/barrier/power_shield/floor
+		to_chat(user, "<span class='notice'>You change the projector to shielded floors.</span>")
+	else
+		holosign_type = /obj/machinery/holosign/barrier/power_shield/wall
+		to_chat(user, "<span class='notice'>You change the projector to shielded walls.</span>")
+
+/obj/item/holosign_creator/powered_shielding/afterattack(atom/target, mob/user, flag)
+	var/turf/T = get_turf(target)
+	var/obj/structure/holosign/H = locate(holosign_type) in T
+	if(H)
+		to_chat(user, "<span class='notice'>You use [src] to deactivate [H].</span>")
+		qdel(H)
+	else
+		if(holocreator_busy)
+			to_chat(user, "<span class='notice'>[src] is busy creating a hologram.</span>")
+			return
+		if(signs.len < max_signs)
+			playsound(src.loc, 'sound/machines/click.ogg', 20, TRUE)
+			if(creation_time)
+				holocreator_busy = TRUE
+				if(!do_after(user, creation_time, target = target))
+					holocreator_busy = FALSE
+					return
+				holocreator_busy = FALSE
+				if(signs.len >= max_signs)
+					return
+			H = new holosign_type(get_turf(target), src)
+			to_chat(user, "<span class='notice'>You create \a [H] with [src].</span>")
+		else
+			to_chat(user, "<span class='notice'>[src] is projecting at max capacity!</span>")
```

Comment:
Use early returns.

Reply:
```if(creation_time)
	holocreator_busy = TRUE
	if(!do_after(user, creation_time, target = target))
		holocreator_busy = FALSE
		return
	holocreator_busy = FALSE
	if(signs.len >= max_signs)
		return```
I can also remove this section since there is no creation time